front/scripts/i18n-checker.ts

```diff
@@ -166,98 +167,174 @@ async function getLocalesKeys(localePath: string, locale: string): Promise<Set<s
   return new Set(allKeys);
 }
 
-// TODO : remove I18nextParser type redefinition once https://github.com/i18next/i18next-parser/issues/1000 gets fixed
-interface I18nextParser {
-  gulp: { new (options: { locales: string[]; output: string }): Transform };
+class I18nSyntaxError extends Error {
+  constructor(file: ts.SourceFile, node: ts.Node, msg: string) {
+    const pos = file.getLineAndCharacterOfPosition(node.pos);
+    super(`${file.fileName}:${pos.line + 1}: ${msg}`);
+  }
 }
 
-/**
- * Scan the source code and generate a locale folder structure
- * in a temp folder, with all the i18n key found.
- *
- * @returns The location of the temp folder
- */
-async function scanCode(): Promise<string> {
-  const tempDir = await mkdtemp(`${os.tmpdir()}/osrd-i18n-`);
-  return new Promise((resolve, reject) => {
-    const stream = vfs
-      .src(`${process.cwd()}/src/**/*.{ts,tsx}`)
-      .pipe(
-        // eslint-disable-next-line new-cap
-        new (I18nextParser as unknown as I18nextParser).gulp({
-          locales: ['dev'],
-          output: '$LOCALE/$NAMESPACE.json',
-        })
-      )
-      .pipe(vfs.dest(tempDir));
-
-    stream.on('finish', () => resolve(tempDir));
-    stream.on('error', (e: Error) => reject(e));
-  });
-}
+function visitCallExpression(
```

Comment:
It would be nice to add some comment to explain a bit the code of this function :)